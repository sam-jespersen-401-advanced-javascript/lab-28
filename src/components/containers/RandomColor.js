import React, { Component } from 'react';
import ColorDisplay from '../colors/ColorDisplay';

let colors = ['red', 'blue', 'pink', 'purple', 'grey', 'orange', 'yellow', 'brown', 'green', 'white', 'black'];
const randomColor = () => colors[Math.floor(Math.random() * colors.length)];

export default class randomColorDisplay extends Component {

    state = {
      backgroundColor: randomColor(),
      backgroundImage: 'none'
    }

    componentDidMount() {

      setInterval(() => {

        const newColor = randomColor();

        if(newColor !== this.state.backgroundColor) {
          this.setState({
            backgroundColor: randomColor(),
            backgroundImage: 'none'
          });

        } else {
          this.setState({ backgroundColor: 'none', backgroundImage: 'url("https://i.dailymail.co.uk/i/pix/2016/03/18/15/324D202500000578-3498922-image-a-33_1458315465874.jpg")' });
        }

      }, 100);
    }

    render() {
      return (
        <ColorDisplay backgroundColor={this.state.backgroundColor} backgroundImage={this.state.backgroundImage} />
      );
    }
}
