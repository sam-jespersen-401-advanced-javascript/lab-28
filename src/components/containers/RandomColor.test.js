import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('ColorContainer module', () => {
  it('renders a color container', () => {
    const wrapper = shallow(<RandomColor />);
    wrapper.setState({ backgroundColor: 'red' });
    expect(wrapper).toMatchSnapshot();
  });
});
