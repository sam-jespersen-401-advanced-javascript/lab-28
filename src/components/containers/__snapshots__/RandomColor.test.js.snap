// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorContainer module renders a color container 1`] = `
<ColorDisplay
  backgroundColor="red"
  backgroundImage="none"
/>
`;
